packages/app/features/Send/InformationInput/components/TransactionFee/AbstractedTransactionFee.tsx

```diff
@@ -0,0 +1,243 @@
+		useSnapshot(injectedElements);
+
+	const {
+		type,
+		token,
+		nftCollection,
+		transactionFee,
+		receiver,
+		sender,
+		amount,
+		nftCollectible,
+		tokenForFee,
+	} = useSnapshot(transactionContext);
+
+	const handleSetTokenFee = (tokenForFee: Token) => {
+		transactionActions.setTokenForFee(tokenForFee as TokenDocument);
+	};
+
+	const dropdownRef = useRef(null);
+
+	let tokenForFeeName = 'Unknown';
+
+	if (network == Networks.solana) {
+		if (tokenForFee && tokenForFee.metadata?.symbol) {
+			tokenForFeeName = tokenForFee.metadata.symbol;
+		} else {
+			tokenForFeeName = 'SOL';
+		}
+	} else if (network == Networks.sui) {
+		tokenForFeeName = 'SUI';
+	} else if (network == Networks.aptos) {
+		tokenForFeeName = 'APT';
+	}
+
+	useEffect(() => {
+		if (!tokenForFee) {
+			transactionActions.setTokenForFee(tokens[0] as TokenDocument);
+		}
+
+		(async () => {
+			const payload: TransactionPayload = {
+				sender: sender,
+				receiver: receiver,
+				tokenForFee: tokenForFee as Token,
+				amount: type === 'Token' ? parseFloat(amount ?? '0') : 1,
+				token:
+					type === 'Token'
+						? (token as TokenDocument)
+						: (nftCollectible as CollectibleDocument),
+				network: token?.network as Networks,
+			};
+
+			if (payload.receiver === '' || payload.amount === 0) {
+				transactionActions.setTransactionFee(0);
+				return;
+			}
+
+			setIsFeeLoading(true);
+			let fee = await getTransactionFee(payload);
+
+			if (tokenForFee?.metadata?.symbol !== 'SOL') {
+				fee = await getTransactionAbstractFee(payload);
+			}
+			transactionActions.setTransactionFee(
+				parseFloat(fee.toPrecision(tokenForFee?.account?.decimals ?? 7)),
+			);
+
+			setIsFeeLoading(false);
+		})();
+	}, [type, token, nftCollection, tokenForFee, receiver, amount]);
+
+	useEffect(() => {
+		if (isDropped) {
+			modalActions.show({
+				id: 'NetworkFee',
+				component: () => (
+					<TokenFeeDropDown
+						tokens={tokens as TokenDocument[]}
+						onSelect={handleSetTokenFee}
+						selectedToken={tokenForFee as Token}
+					/>
+				),
+				bindingRef: dropdownRef,
+				bindingDirection: BindDirections.Bottom,
+				maskActiveOpacity: 0,
+			});
+		}
+	}, [isDropped]);
+
+	useEffect(() => {
+		if (
+			parseFloat(tokenForFee?.account.balance as string) /
+				10 ** (tokenForFee?.account.decimals ?? 0) <
+			(transactionFee ?? 0)
+		) {
+			setError(
+				`Not enough ${
+					tokenForFee?.metadata?.symbol ?? 'Unknown'
+				}, select other token`,
+			);
+		} else {
+			setError('');
+		}
+	}, [transactionFee]);
+
+	return (
+		<View>
+			<View style={styles.container}>
+				<View style={styles.titleContainer}>
+					<Exclamation color="#566674" size={10} />
+					<Text style={styles.titleText}>Transaction fee</Text>
+				</View>
+
+				<View style={styles.valueContainer}>
+					{isFeeLoading ? (
+						<ActivityIndicator size="small" color="#FFFFFF" />
+					) : (
+						<Text
+							style={[styles.feeText, error !== '' && { color: '#FC9B0A' }]}
```

Comment:
can it be checked like this `!err && {}`?

Reply:
no, if i write like that, it does not render as i want it to